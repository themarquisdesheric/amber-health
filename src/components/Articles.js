import React from 'react';
import PropTypes from 'prop-types';
import _kebabCase from 'lodash/kebabCase';
import { Link, graphql, StaticQuery } from 'gatsby';
import Breadcrumbs from './Breadcrumbs';
import Article from './Article';

const completedPost = ({ node }) => !node.frontmatter.draft;

const topics = ['Ehlers-Danlos Syndrome', 'Interstitial Cystitis', 'Lichen Sclerosus', 'Endometriosis'];
// TODO: once topics are ready
const Topics = () => (
  <div>
    <p className="text-center text-md font-bold mt-8 py-3">Topics</p>
    <ul className="topics">
      {topics.map(topic => 
        <Link to={`/tags/${_kebabCase(topic)}/`} className="text-sm relative pl-5" key={topic}>
          {` ${topic}`}
        </Link>
      )}
    </ul>
  </div>
);

export const Articles = ({ data }) => {
  const { edges: posts = [] } = data.allMarkdownRemark;
  
  return (
    <section className="content">
      <Breadcrumbs path="Articles" />
      <div className="flex flex-col items-center">
        {posts.filter(completedPost).map(({ node: post }) => 
          <Article post={post} className="article-card max-w-xl" key={post.id} />
        )}
        {/* <Topics /> */}
      </div>
    </section>
  );
};

Articles.propTypes = {
  data: PropTypes.shape({
    allMarkdownRemark: PropTypes.shape({
      edges: PropTypes.array
    })
  })
};

export default () => (
  <StaticQuery
    query={graphql`
      query ArticlesQuery {
        allMarkdownRemark(
          sort: { order: DESC, fields: [frontmatter___date] }
          filter: { frontmatter: { templateKey: { eq: "article" } } }
        ) {
          edges {
            node {
              id
              fields {
                slug
              }
              frontmatter {
                title
                description
                draft
                templateKey
                featuredimage {
                  childImageSharp {
                    fluid(maxWidth: 576, quality: 100) {
                      ...GatsbyImageSharpFluid
                    }
                  }
                }
              }
            }
          }
        }
      }
    `}
    render={(data, count) => <Articles data={data} count={count} />}
  />
);
