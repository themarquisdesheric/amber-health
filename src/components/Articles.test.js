import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Articles } from './Articles';

const post = {
  node: {
    id: 'a3a3d5f0-fd76-5738-9d6c-b6e250ee9f10',
    fields: {
      slug: '/articles/2019-09-27-endometriosis-diagnosis/'
    },
    frontmatter: {
      title: 'Endometriosis Diagnosis',
      description: 'Learn how endometriosis is diagnosed',
      draft: false,
      templateKey: 'article',
      featuredimage: {
        childImageSharp: {
          fluid: {
            base64: 'base64',
            aspectRatio: 1.5,
            src: '/static/a162fe840d8c214c93ae37d3e6fc212a/6806c/pom-2.jpg',
            srcSet: 'srcSet',
            sizes: '(max-width: 576px) 100vw, 576px'
          }
        }
      }
    }
  }
};

const secondPost = {
  node: {
    ...post.node,
    frontmatter: {
      ...post.node.frontmatter
    }
  }
};

const data = {
  allMarkdownRemark: {
    edges: [
      post,
      secondPost
    ]
  }
};

describe('Articles', () => {
  let mockData;

  beforeEach(() => {
    mockData = data;
  });

  test('should match snapshot', () => {
    const wrapper = shallow(<Articles data={mockData} />);
    
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('Article').length).toBe(2);
  });
  
  test('should not render draft posts', () => {
    mockData.allMarkdownRemark.edges[1].node.frontmatter.draft = true;

    const wrapper = shallow(<Articles data={mockData} />);

    expect(wrapper.find('Article').length).toBe(1);
  });

  test('should render draft posts if GATSBY_SHOW_DRAFTS is truthy', () => {
    process.env.GATSBY_SHOW_DRAFTS = true;

    mockData.allMarkdownRemark.edges[1].node.frontmatter.draft = true;

    const wrapper = shallow(<Articles data={mockData} />);

    expect(wrapper.find('Article').length).toBe(2);
  });
});
