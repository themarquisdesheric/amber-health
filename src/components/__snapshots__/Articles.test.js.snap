// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles should match snapshot 1`] = `
<section
  className="content"
>
  <Breadcrumbs
    path="Articles"
  />
  <div
    className="flex flex-col items-center"
  >
    <Article
      className="article-card max-w-xl"
      key="a3a3d5f0-fd76-5738-9d6c-b6e250ee9f10"
      post={
        Object {
          "fields": Object {
            "slug": "/articles/2019-09-27-endometriosis-diagnosis/",
          },
          "frontmatter": Object {
            "description": "Learn how endometriosis is diagnosed",
            "draft": false,
            "featuredimage": Object {
              "childImageSharp": Object {
                "fluid": Object {
                  "aspectRatio": 1.5,
                  "base64": "base64",
                  "sizes": "(max-width: 576px) 100vw, 576px",
                  "src": "/static/a162fe840d8c214c93ae37d3e6fc212a/6806c/pom-2.jpg",
                  "srcSet": "srcSet",
                },
              },
            },
            "templateKey": "article",
            "title": "Endometriosis Diagnosis",
          },
          "id": "a3a3d5f0-fd76-5738-9d6c-b6e250ee9f10",
        }
      }
    />
    <Article
      className="article-card max-w-xl"
      key="a3a3d5f0-fd76-5738-9d6c-b6e250ee9f10"
      post={
        Object {
          "fields": Object {
            "slug": "/articles/2019-09-27-endometriosis-diagnosis/",
          },
          "frontmatter": Object {
            "description": "Learn how endometriosis is diagnosed",
            "draft": false,
            "featuredimage": Object {
              "childImageSharp": Object {
                "fluid": Object {
                  "aspectRatio": 1.5,
                  "base64": "base64",
                  "sizes": "(max-width: 576px) 100vw, 576px",
                  "src": "/static/a162fe840d8c214c93ae37d3e6fc212a/6806c/pom-2.jpg",
                  "srcSet": "srcSet",
                },
              },
            },
            "templateKey": "article",
            "title": "Endometriosis Diagnosis",
          },
          "id": "a3a3d5f0-fd76-5738-9d6c-b6e250ee9f10",
        }
      }
    />
  </div>
</section>
`;
